src/app/armory/Links.tsx

```diff
@@ -149,26 +127,48 @@ function buildLightGGSockets(item: DimItem) {
  * D2Gunsmith's socket format is: [...<first four perks, padded out if necessary, masterwork, weapon mod].join(',')
  */
 function buildGunsmithSockets(item: DimItem) {
-  if (item.sockets) {
-    const perkValues: number[] = [0, 0, 0, 0];
+  const perkValues = getWeaponSocketInfo(item);
+
+  if (perkValues) {
+    return `?p=${perkValues.traits.join(',')},${perkValues.masterwork},${perkValues.weaponMod}`;
+  }
+
+  return '';
+}
+
+/**
+ * Gathers general socket information for link generation in D2Gunsmith and Light.gg.
+ */
+function getWeaponSocketInfo(
+  item: DimItem
+): null | { traits: number[]; originTrait: number; masterwork: number; weaponMod: number } {
+  if (item.sockets && item.bucket?.inWeapons) {
+    const traits: number[] = [0, 0, 0, 0];
     const perks = getSocketsWithStyle(item.sockets, DestinySocketCategoryStyle.Reusable);
     perks.unshift(); // remove the archetype perk
     let i = 0;
-    // Only pick 4 perks and purposefully ignore the origin perk
-    // on some guns as the link format doesn't support them.
+    // First 4 perks are on all weapons
     for (const perk of _.take(perks, 4)) {
-      perkValues[i] = perk.plugged?.plugDef.hash ?? 0;
+      traits[i] = perk.plugged?.plugDef.hash ?? 0;
       i++;
     }
-    const masterwork = item.sockets.allSockets.find(isWeaponMasterworkSocket);
-    perkValues[4] = masterwork?.plugged?.plugDef.hash ?? 0;
-    const weaponMod = item.sockets.allSockets.find((s) =>
+
+    const origin = item.sockets.allSockets.find((s) =>
+      s.plugged?.plugDef.itemCategoryHashes?.includes(ItemCategoryHashes.WeaponModsOriginTraits)
+    );
+    const originTrait = origin?.plugged!.plugDef.hash ?? 0;
+
+    // TODO: Map enhanced intrinsic frames with their corresponding stat masterworks
+    const masterworkSocket = item.sockets.allSockets.find(isWeaponMasterworkSocket);
+    const masterwork = item?.crafted ? 0 : masterworkSocket?.plugged?.plugDef.hash ?? 0;
```

Comment:
Hmm yeah, Armory essentially takes the sockets from the actual item and applies them to a fake item to prevent messing with the actual item, which bypasses a setting a bunch of precalculated info. This should catch it.

```suggestion
    const masterwork =
      masterworkSocket?.plugged?.plugDef.plug.plugCategoryHash ===
      PlugCategoryHashes.CraftingPlugsFrameIdentifiers
        ? 0
        : masterworkSocket?.plugged?.plugDef.hash ?? 0;
```